### PR TITLE
Don't duplicate CrossJava logic

### PR DESCRIPTION
**Pull request by raboof**
As we can now take that from upstream sbt 1.3.x

Refs #25830

### CODE REVIEW

**Review by patriknw — APPROVED** [review]
LGTM, I guess

File: project/JdkOptions.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      jdk8options
+    else fullJavaHomes.get("8") match {
+      case Some(java8home) =>
+        jdk11options(java8home)
+      case None =>
+        throw new MessageOnlyException("A JDK 8 installation was not found, but is required to build Akka. To target your system JDK, use the \"set every targetSystemJdk := true\" sbt command. Resulting artifacts may not work on JDK 8")
+    }
+
+  val targetJdkSettings = Seq(
+    targetSystemJdk := false,
```
Comment (helena) [inline project/JdkOptions.scala]: I'd expect that trailing comma to break the build?

**Review by helena — APPROVED** [review]
Nice and clean, LGTM

File: project/JdkOptions.scala (hunk 1 above)
Comment (raboof) [reply project/JdkOptions.scala]: I think we don't check the formatting of the sbt configuration

**Pull request merged**